### PR TITLE
 Add support for IPv6 addresses with the google_compute_external_vpn_gateway terraform resource

### PR DESCRIPTION
**Pull request by abhijeetkjha-google**
```release-note:enhancement
compute: added `interface.ipv6-address` field in `google_compute_external_vpn_gateway` resource
```

As per the design doc, [go/arcus-outer-vpn-ipv6](https://goto.google.com/arcus-outer-vpn-ipv6). The support for IPv6 is present for Outer HA VPN, and support for IPV6 address is present in the api implementation and gcloud for external_vpn_gateway. I see the field, ipv6_address which is the IPv6 address of the interface in the external VPN gateway, Is missing in the terraform implementation for external vpn gateway. Adding the missing ipv6_address field in the yaml file.

### CODE REVIEW

**Comment by abhijeetkjha-google** [comment]
Tested locally on the unit and lint tests.

File: mmv1/third_party/terraform/services/compute/resource_compute_external_vpn_gateway_test.go [hunk 1]
```diff
@@ -65,3 +65,42 @@ resource "google_compute_external_vpn_gateway" "external_gateway" {
...
+func computeExternalVPNGatewayIpv6AddressConfig(suffix, ipv6_address string) string {
+	return fmt.Sprintf(`
+resource "google_compute_external_vpn_gateway" "external_gateway" {
+  name            = "tf-test-external-gateway-%s"
+  redundancy_type = "SINGLE_IP_INTERNALLY_REDUNDANT"
+  description     = "An externally managed VPN gateway"
+  interface = [{
+    id         = 0
+    ipv6_address = "%s"
+  }]
```
Comment (NickElliot) [inline mmv1/third_party/terraform/services/compute/resource_compute_external_vpn_gateway_test.go:103]:
```suggestion
  interface {
    id         = 0
    ipv6_address = "%s"
  }
```

**Review by NickElliot — CHANGES_REQUESTED** [review]
minor edit for syntax on the test otherwise lgtm

File: mmv1/third_party/terraform/services/compute/resource_compute_external_vpn_gateway_test.go (hunk 1 above)
Comment (abhijeetkjha-google) [reply mmv1/third_party/terraform/services/compute/resource_compute_external_vpn_gateway_test.go]: Done. Thanks.

**Comment by abhijeetkjha-google** [comment]
> minor edit for syntax on the test otherwise lgtm

Fixed.

**Review by abhijeetkjha-google — COMMENTED** [review]
Added the requested change.

**Review by NickElliot — APPROVED** [review]
unrelated failure, LGTM!

**Pull request merged**